subprojects/core-api/src/main/java/org/gradle/api/Project.java

```diff
@@ -1316,33 +1316,44 @@ public interface Project extends Comparable<Project>, ExtensionAware, PluginAwar
     void allprojects(@DelegatesTo(Project.class) Closure configureClosure);
 
     /**
-     * Adds an action to execute immediately before this project is evaluated.
+     * <p>Adds an action to call immediately before this project is evaluated. Passes the project to the
+     * action as a parameter. Actions passed to this method execute in the same order they were passed.</p>
+     * 
+     * <p>If the project has already been evaluated, the action never executes.</p>
+     * <p>If you call this method within a <code>beforeEvaluate</code> action, the passed action executes after all
+     * previously added <code>beforeEvaluate</code> actions finish executing.</p>
      *
      * @param action the action to execute.
      */
     void beforeEvaluate(Action<? super Project> action);
 
     /**
-     * Adds an action to execute immediately after this project is evaluated.
+     * <p>Adds an action to call immediately after this project is evaluated. Passes the project to the
```

Comment:
Can we add a paragraph break after the first sentence so that the DSL ref doesn't get too crowded:
![image](https://user-images.githubusercontent.com/5387972/181392711-48aed949-0d0f-4006-97f5-ddaa134e08f3.png)